packages/@aws-cdk/aws-ec2/lib/vpc.ts

```diff
@@ -1152,7 +1168,9 @@ export class Vpc extends VpcBase {
 
     this.vpcId = this.resource.ref;
 
-    this.subnetConfiguration = ifUndefined(props.subnetConfiguration, Vpc.DEFAULT_SUBNETS);
+    // If natGateways = 0 ==> no private subnets, made them isolated instead
```

Comment:
nit: I think your code is self-documenting (with a command-click or two) and this comment is not really needed.